### PR TITLE
Minor changes for better coding style.

### PR DESCRIPTION
**Pull request by DanjieTang**

### CODE REVIEW

File: system_config.yaml [hunk 1]
```diff
@@ -34,7 +34,7 @@ ANSWER_MULT_QS_FORMAT_RESP_PROMPT: "format_mult_qs_prompt.jinja"
 PATH_TO_DOMAIN_CONFIGS: "domain_specific/configs/restaurant_configs"
 MODEL: "gpt-3.5-turbo"
 SEARCH_ENGINE: "vector database"
-ENABLE_MULTITHREADING: True
+ENABLE_MULTITHREADING: False
 UNACCEPTABLE_SIMILARITY_SCORE_RANGE: 0.5
```
Comment (kathy23980) [inline system_config.yaml:38]: is there a reason why you changed this? if not, just set it to True

File: domain_specific/classes/restaurants/geocoding/nominatim_wrapper.py [hunk 2]
```diff
@@ -8,8 +8,11 @@ class NominatimWrapper(GeocoderWrapper):
     Wrapper for Nominatim geocoder.
     """
 
+    _geocoder: Nominatim
+    _mandatory_address_key: dict[str]
     _geocoder_history: dict[str, Location]
```
Comment (kathy23980) [inline domain_specific/classes/restaurants/geocoding/nominatim_wrapper.py:13]: the type of _mandatory_address_key is str

File: domain_specific/classes/restaurants/geocoding/google_v3_wrapper.py [hunk 3]
```diff
@@ -14,6 +14,8 @@ class GoogleV3Wrapper(GeocoderWrapper):
     :param mandatory_address_keys: key used to determine if location is specific enough
     """
 
+    _geocoder: GoogleV3
+    _mandatory_address_keys: dict[str]
     _geocoder_history: dict[str, Location]
```
Comment (kathy23980) [inline domain_specific/classes/restaurants/geocoding/google_v3_wrapper.py:19]: the type of _mandatory_address_keys is set

File: domain_specific/classes/restaurants/location_constraint_merger.py [hunk 4]
```diff
@@ -2,25 +2,32 @@
...
-
 class LocationConstraintMerger(ConstraintMerger):
+
+    """
+    Merge the new and old location constraints
+
+    :param _geocoder_wrapper: Wrapper for geocoding
+    """
+    _geocoder_wrapper: GeocoderWrapper
+
```
Comment (kathy23980) [inline domain_specific/classes/restaurants/location_constraint_merger.py:13]: add _constraint: str

File: domain_specific/classes/restaurants/location_status.py [hunk 5]
```diff
@@ -6,6 +6,8 @@
 class LocationStatus(ConstraintStatus):
     """
     Class representing the status for location constraint
+
+    :param _geocoder_wrapper: Wrapper for geocoding
     """
```
Comment (kathy23980) [inline domain_specific/classes/restaurants/location_status.py:11]: here param is parameters for init, so should be geocoder_wrapper

File: rec_action/response_type/recommend_resp.py [hunk 6]
```diff
@@ -7,6 +7,10 @@ class RecommendResponse(Response):
 
     :param domain: domain of the recommendation (e.g. restaurants)
     """
+
+    _current_recommended_items: list
```
Comment (kathy23980) [inline rec_action/response_type/recommend_resp.py:11]: use list[RecommendedItem] if possible. also add return type to get_current_recommended_items function

**Review by kathy23980 — CHANGES_REQUESTED** [review]
also remove current item extractor from accept_recommendation.py and change template attribute to be private for each user intent.

File: system_config.yaml (hunk 1 above)
Comment (DanjieTang) [reply system_config.yaml:38]: Oh, sorry. That was my mistake, I'll fix it.

File: state/constraints/constraint_merger.py [hunk 7]
```diff
@@ -17,7 +19,7 @@ def get_constraint(self) -> str:
         """
         return self._constraint
 
-    def merge_constraint(self, og_constraint_value, new_constraint_value) -> None:
+    def merge_constraint(self, og_constraint_value: list[str], new_constraint_value: list[str]) -> None:
         """
```
Comment (kathy23980) [inline state/constraints/constraint_merger.py:23]: return type should be list[str]

**Review by kathy23980 — COMMENTED** [review]
Can you also rename _item_ids to be _item_indices and _get_item_ids() to be _get_item_indices() in recommend_prompt_based_resp?

**Review by kathy23980 — APPROVED** [review]
Looks good.

**Pull request merged**